app/domain/authentication/resource_restrictions/get_restrictions_from_annotation.rb

```diff
@@ -0,0 +1,48 @@
+require 'command_class'
+
+module Authentication
+  module ResourceRestrictions
+    # Parses the given annotation name and tries to extract a restriction from it.
+    # A restriction can have 2 formats:
+    # 1. "<Authenticator name>/<Restriction name>"
+    # 2. "<Authenticator name>/<Service ID>/<Restriction name>"
+    #
+    # The first format is a general restriction.
+    # This means it is relevant to all services of this authenticator.
+    #
+    # The second format is specific to the specified service ID.
+    # This means all other services will ignore it.
+    # Moreover, if both formats are found for the same restriction, the specific one will be used.
+    #
+    # This function returns the restriction name, if found (nil otherwise) and a
+    # boolean indicating if it is a general restriction or not.
+    GetRestrictionsFromAnnotation = CommandClass.new(
```

Comment:
This command is singular, not plural.
It retrieves single restriction.
```suggestion
    GetRestrictionFromAnnotation = CommandClass.new(
```

Reply:
Remember to rename file name

Reply:
Done